feat: added password change component

alvgonfri opened the pull request:
This pull request introduces a change password component that includes a form to change the password of the logged-in user. This component can be easily added to users' profile pages.

> [!NOTE]  
> To test this you must temporarily include the component on any of the application pages.

alvgonfri committed:
feat: added password change component

TunixR commented on src/forms/auth/schemas.js:
```diff
@@ -16,3 +16,12 @@ export const verify2FASchema = z.object({
     .transform((val) => val.trim()),
 });
+
+export const changePasswordSchema = z.object({
+  currentPassword: z
+    .string({ required_error: 'Current password is required' })
+    .transform((val) => val.trim()),
+  newPassword: z
+    .string({ required_error: 'New password is required' })
+    .transform((val) => val.trim()),
+});
```
```suggestion
export const changePasswordSchema = z.object({
  currentPassword: z
    .string({ required_error: 'Current password is required' })
    .transform((val) => val.trim()),
  newPassword: z
    .string({ required_error: 'New password is required' })
    .transform((val) => val.trim()),
  confirmPassword: z
    .string({ required_error: 'You need to confirm password' })
    .transform((val) => val.trim()),
}).refine((data) => data.newPassword === data.confirmPassword, {
    message: "Passwords don't match",
    path: ["confirmPassword"],
});

TunixR commented on src/forms/auth/forms.jsx:
```diff
@@ -100,3 +100,53 @@ export function Verify2FAForm({ form, onSubmit, isLoading, error }) {
+            </FormItem>
+          )}
+        />
+        <FormField
+          control={form.control}
+          name="newPassword"
+          rules={{ required: true }}
+          render={({ field }) => (
+            <FormItem className="flex flex-col items-start">
+              <FormLabel className="text-base font-medium">New Password</FormLabel>
+              <FormControl>
+                <PasswordInput {...field} />
+              </FormControl>
+              {form.formState.errors.newPassword && (
+                <FormMessage>{form.formState.errors.newPassword.message}</FormMessage>
+              )}
+            </FormItem>
+          )}
+        />
+        <Button className="w-full" type="submit" disabled={isLoading}>
+          {isLoading && (
+            <LoaderCircle className="mr-2 h-4 w-4 animate-spin" />
+          )}
+          <span className="text-base font-medium">Update Password</span>
+        </Button>
```
```suggestion
        <FormField
          control={form.control}
          name="newPassword"
          rules={{ required: true }}
          render={({ field }) => (
            <FormItem className="flex flex-col items-start">
              <FormLabel className="text-base font-medium">New Password</FormLabel>
              <FormControl>
                <PasswordInput {...field} />
              </FormControl>
              {form.formState.errors.newPassword && (
                <FormMessage>{form.formState.errors.newPassword.message}</FormMessage>
              )}
            </FormItem>
          )}
        />
        <FormField
          control={form.control}
          name="confirmPassword"
          rules={{ required: true }}
          render={({ field }) => (
            <FormItem className="flex flex-col items-start">
              <FormLabel className="text-base font-medium">Confirm Password</FormLabel>
              <FormControl>
                <PasswordInput {...field} />
              </FormControl>
              {form.formState.errors.confirmPassword && (
                <FormMessage>{form.formState.errors.confirmPassword.message}</FormMessage>
              )}
            </FormItem>
          )}
        />
        <Button className="w-full" type="submit" disabled={isLoading}>
          {isLoading && (
            <LoaderCircle className="mr-2 h-4 w-4 animate-spin" />
          )}
          <span className="text-base font-medium">Update Password</span>
        </Button>
```

TunixR commented on src/components/forms/password-change.jsx:
```diff
@@ -0,0 +1,86 @@
+
+  const form = useForm({
+    resolver: zodResolver(changePasswordSchema),
+    defaultValues: {
+      currentPassword: '',
+      newPassword: '',
+    },
```
```suggestion
    defaultValues: {
      currentPassword: '',
      newPassword: '',
      confirmPassword:'',
},

TunixR commented on src/components/forms/password-change.jsx:
```diff
@@ -0,0 +1,86 @@
+  const onSubmit = async (values) => {
+    setIsLoading(true);
+    setError('');
+    changePassword(values)
```
Here only provide current and new password

TunixR reviewed (changes requested):
Very good overall, but I suggest adding a confirmPassword field

alvgonfri committed:
feat: added confirm password field

alvgonfri commented:
Good suggestion!

I added the password confirmation field, but doing so made the dropdown too small for the form, so I changed it to a modal. Making the change to modal has also resolved the problem that did not allow tabulation between form fields.

Tell me if you see it well or do you think we should go back to the dropdown form.

TunixR reviewed (approved):
LGTM
